pkgs/tools/networking/networkmanager/default.nix

```diff
@@ -118,8 +118,6 @@ stdenv.mkDerivation (finalAttrs: {
     # almost cross-compiles, however fails with
     # ** (process:9234): WARNING **: Failed to load shared library '/nix/store/...-networkmanager-aarch64-unknown-linux-gnu-1.38.2/lib/libnm.so.0' referenced by the typelib: /nix/store/...-networkmanager-aarch64-unknown-linux-gnu-1.38.2/lib/libnm.so.0: cannot open shared object file: No such file or directory
     "-Ddocs=${lib.boolToString (stdenv.buildPlatform == stdenv.hostPlatform)}"
-    # We don't use firewalld in NixOS
-    "-Dfirewalld_zone=false"
```

Comment:
Q: does this have any effect on users using networkmanager who aren't using firewalld?

Reply:
If you [search this word](https://gitlab.freedesktop.org/search?search=firewalld_zone&nav_source=navbar&project_id=411&group_id=1204&search_code=true&repository_ref=main), you can find that this only does 2 things:

1. install `$out/lib/firewalld/zones/nm-shared.xml`
2. possibly set default zone to `nm-shared` https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/core/devices/nm-device.c#L13316-13324

Since NetworkManager zones are essentially tied to firewalld zones, it won't have any effect if you don't have firewalld enabled.

Reply:
Hey there, my PC is an `x86_64-linux` machine and this PR did in fact cause some rebuilds for my configuration. I just let it build for a few hours on max jobs set to 4 and it went mostly smoothly.

Seeing your comment for the `x86_64-linux`, I remember one package stalling in its test suite. I don't remember which package it was, but I think I just tried exiting and restarting the build for that package a few times. I know the tests were related to HTTP requests, but I think it was probably due to some tests being flaky because I was still running my iptables-based config.

Reply:
I'll open a separate PR about NetworkManager.